### ISSUE TITLE
BCML fails to install...

### ISSUE THREAD

**Issue by BigheadSMZ** [NiceneNerd/BCML#315]
Over the years I have grown to hate Python. Every time I come across a project written in Python, I can be guaranteed to face issues because I didn't have some requirement, or I'm using a version that is 0.0.1 difference from the one I was supposed to be using, or something stopped building for some reason, some server is down, etc. etc. And just like every other Python project I've ever come across, this simply does not want to work. I installed the Visual C++ requirement but still... trying to install it just spills a bunch of error messages across the screen, which can be found below.

https://pastebin.com/FkvGPsmm

Sorry if I seem ungrateful for the effort, I'm just frustrated because this seems to be a regular occurrence. I really don't like Python. The fault probably lies entirely with me, but the fact I have so many problems getting Python projects to actually work, more times than projects written in every other language combined, it has made me bitter towards it. Using Windows 11 fully updated, Python v3.9.9, every C++ redistributable installed from the past 2 decades. I have no clue why it's not working...

![image](https://user-images.githubusercontent.com/9309452/166100001-033946b7-7f1f-4def-b5b9-17421ec8f501.png)

**Comment by HGStone** [comment]
i think you should rename to the title of this issue to BigheadSMZ fails to read...

anyways.. al the instructions clearly state only python 3..7.x or 3.8.x are supported, yet you install python 3.9.x

I suggest uninstalling all your python versions and then installing python 3.8.10 64bit downloaded from python.org and make sure you check the ADD TO PATH box

after that type `pip install bcml cefpython3 tk`

then after it installs just type `bcml` at a cmd prompt

**Comment by BigheadSMZ** [comment]
No you're right, that was definitely me being stupid. I can't deny that I failed to read that part and just made assumptions, even knowing the problem all along was probably exactly this: a wrong version. My brain turned off after reading 3.7+ in the instructions because I was just trying to get it up and running as fast as possible while also working on other things. My focus wasn't entirely here and I jumped to conclusions, and instead of going back and reading the instructions I came here to rant about the error messages. Usually when it comes to anything else, install past the minimum required version, and you're all set for the future. But not with Python... this really is the main reason why it makes me crazy.

I didn't install 3.9 for this, and I can't just uninstall it just to use this. I've had 3.9 installed for awhile now because a few other scripts/projects that I use regularly requires that version, but does not work in 3.10 or 3.8. Another script I run, also breaks on 3.10, so 3.9 works for both of those. My own project utilizes some python scripts for lz4 compression/decompression, and specifically requires 3.9 (because Python reasons). So BCML requires 3.7 or 3.8, but does not work with 3.9. So now if I want to use it, I'll have to install multiple Python versions which is I guess a minor inconvenience but still one none the less. And yes, I already have both the Python base folder and the scripts folder on "path", so now I have to get creative with a few batch scripts and just launch them from the installation folder since only one can be on "path" at a time (unless renaming the executables works, but that sounds...like a bad workaround).

When compatibility of .py scripts and modules is broken every other minor version, it gets really old really fast. For example, Java code written in the late 90s should still work today. I can write a somewhat complex PowerShell script for v1.0, and it still works in v7.0 that has 16 years of development between them. Most batch scripts written for DOS can still work in the Windows command prompt. Then there's Python.. where scripts are lucky to keep working between minor revisions, as this situation exemplifies. I guess its nice if you only use it for one specific task, but as soon as you have multiple projects that rely on it, requiring multiple versions.. it just gets annoying to deal with. 

I do apologize for coming off as ignorant and rude, my disdain for Python clouds my judgement. I know a lot of people like it, I know some really great things are written in it, and from what I can tell its really easy to pick up... I just never got into it because of the numerous compatibility issues I've experienced over the years. But yeah, I didn't come here just so i could complain about it. I do apologize for going off on this tangent and thank you for taking the time to help despite my complaints and inability to read past a single sentence.
